src/include/migraphx/shape_for_each.hpp

```diff
@@ -51,9 +51,13 @@ void shape_for_each(const migraphx::shape& s, F f)
                            assert(len > 0 and stride > 0);
                            return (i / stride) % len;
                        });
-        call(indices);
+        if constexpr(std::is_invocable_v<F, Vec, Idx>)
```

Comment:
Do `std::is_invocable<F, decltype(index_const_ref), decltype(i)>{}` instead. Use `is_invocable` instead of `is_invocable_v` and use `decltype` instead of adding template parameters.

Reply:
Good idea. Thanks. Done.